GoogleForCreators/web-stories-wp
# Media Recording: add device selection

## Context



In case you have multiple mics or cameras, we want to allow you choosing the right one

## Summary



This PR adds 2 drop-down menus to select microphone and camera as per the information received by the browser.

**Note: This just contains the functionality and is not pretty since design is subject to change**

## Relevant Technical Choices



## To-do



In the future:

* Update UI and user flow after getting new designs
* Improve error handling (also relies on design)

## User-facing changes



Not pretty, but gets the job done:

<img width="567" alt="Screenshot 2022-04-28 at 12 04 54" src="https://user-images.githubusercontent.com/841956/165729968-d4746682-1e8d-47fa-8100-7569d2ed9e87.png">

## Testing Instructions






This PR can be tested by following these steps:

1. Start media recording
2. Switch microphone/camera
3. Camera/microphone used for recording should change accordingly


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #11078
